### PR TITLE
12468 disallow double underscores in custom field names

### LINKED ISSUE #12468: Disallow the use of double underscores in a custom field name
### NetBox version

v3.5.0

### Python version

3.8

### Steps to Reproduce

1. Create Custom Field --> Type: Object (Location), Assigned Models: ipam | VLAN
2. Go to IPAM --> Vlans --> Edit Vlan and chose Location Object by custom Field
3. In the Vlans overview chosed location not showing, if you click on the specific vlan the location is showing
4. Same behavior if you assign the custom field to other models like ipam | ip addresses

### Expected Behavior

Location from custom Field should be shown in vlan overview

### Observed Behavior

Location from custom Field no shown in vlan overview
Netbox_Location_not_show

Location from custom Field only show if you click on the vlan
Netbox_Location_show_in_vlan

### PR DESCRIPTION
**Pull request by arthanson**
### Fixes: #12468 

Disallow double-underscore in custom field names.  Unfortunately creates a migration, Jeremy what is the official way to suppress this migration?

### CODE REVIEW

**Comment by jeremystretch** [comment]
For model field changes that don't actually modify the database schema (such as modifying application-level validators as in this case), we can just tweak the most recent definition of the field in the existing migrations. This happens to be [0066_customfield_name_validation](https://github.com/netbox-community/netbox/blob/develop/netbox/extras/migrations/0066_customfield_name_validation.py).

Also: Rather than modifying the existing regex, it might be cleaner to add a second regex validator to check specifically for double underscores. This will allow us to use simpler expressions and to provide separate error messages to the user.

**Comment by arthanson** [comment]
@jeremystretch updated.

File: netbox/extras/models/customfields.py [hunk 1]
```diff
@@ -85,6 +85,11 @@ class CustomField(CloningMixin, ExportTemplatesMixin, ChangeLoggedModel):
                 message="Only alphanumeric characters and underscores are allowed.",
                 flags=re.IGNORECASE
             ),
+            RegexValidator(
+                regex=r'^(?!.*__.*)',
```
[the remark this answers is not in the document]
Comment (jeremystretch) [reply netbox/extras/models/customfields.py:89]: My mistake, I forgot to call out the need to set `inverse_match=True` on the validator. With that, the simpler regex should suffice.

[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/extras/models/customfields.py]: doh. Thanks for pointing that out

**Pull request merged**